// @circle-fin/circle-sdk/dist/generated/models/payment-error-code.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentErrorCode = void 0;
/**
 * Indicates the failure reason of a payment. Only present for payments in failed state. Possible values are [`payment_failed`, `payment_fraud_detected`, `payment_denied`, `payment_not_supported_by_issuer`, `payment_not_funded`, `payment_unprocessable`, `payment_stopped_by_issuer`, `payment_canceled`, `payment_returned`, `payment_failed_balance_check`, `card_failed`, `card_invalid`, `card_address_mismatch`, `card_zip_mismatch`, `card_cvv_invalid`, `card_expired`, `card_limit_violated`, `card_not_honored`, `card_cvv_required`, `credit_card_not_allowed`, `card_account_ineligible`, `card_network_unsupported`, `channel_invalid`, `unauthorized_transaction`, `bank_account_ineligible`, `bank_transaction_error`, `invalid_account_number`, `invalid_wire_rtn`, `invalid_ach_rtn`, `vendor_inactive`]\'
 * @export
 * @enum {string}
 */
exports.PaymentErrorCode = {
    PaymentFailed: "payment_failed",
    PaymentFraudDetected: "payment_fraud_detected",
    PaymentDenied: "payment_denied",
    PaymentNotSupportedByIssuer: "payment_not_supported_by_issuer",
    PaymentNotFunded: "payment_not_funded",
    PaymentUnprocessable: "payment_unprocessable",
    PaymentStoppedByIssuer: "payment_stopped_by_issuer",
    PaymentCanceled: "payment_canceled",
    PaymentReturned: "payment_returned",
    PaymentFailedBalanceCheck: "payment_failed_balance_check",
    CardFailed: "card_failed",
    CardInvalid: "card_invalid",
    CardAddressMismatch: "card_address_mismatch",
    CardZipMismatch: "card_zip_mismatch",
    CardCvvInvalid: "card_cvv_invalid",
    CardExpired: "card_expired",
    CardLimitViolated: "card_limit_violated",
    CardNotHonored: "card_not_honored",
    CardCvvRequired: "card_cvv_required",
    CardRestricted: "card_restricted,",
    CardAccountIneligible: "card_account_ineligible",
    CardNetworkUnsupported: "card_network_unsupported",
    ChannelInvalid: "channel_invalid",
    UnauthorizedTransaction: "unauthorized_transaction",
    BankAccountIneligible: "bank_account_ineligible",
    BankTransactionError: "bank_transaction_error",
    InvalidAccountNumber: "invalid_account_number",
    InvalidWireRtn: "invalid_wire_rtn",
    InvalidAchRtn: "invalid_ach_rtn",
    RefIdInvalid: "ref_id_invalid",
    AccountNameMismatch: "account_name_mismatch",
    AccountNumberMismatch: "account_number_mismatch",
    AccountIneligible: "account_ineligible",
    WalletAddressMismatch: "wallet_address_mismatch",
    CustomerNameMismatch: "customer_name_mismatch",
    InstitutionNameMismatch: "institution_name_mismatch",
    VendorInactive: "vendor_inactive"
};
