// @circle-fin/circle-sdk/dist/generated/models/payout-error-code.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PayoutErrorCode = void 0;
/**
 * Indicates the failure reason of a payout. Only present for payouts in failed state. Possible values are [`insufficient_funds`, `transaction_denied`, `transaction_failed`, `transaction_returned`, `bank_transaction_error`, `fiat_account_limit_exceeded`, `invalid_bank_account_number`, `invalid_ach_rtn`, `invalid_wire_rtn`, `vendor_inactive`]\'
 * @export
 * @enum {string}
 */
exports.PayoutErrorCode = {
    InsufficientFunds: "insufficient_funds",
    TransactionDenied: "transaction_denied",
    TransactionFailed: "transaction_failed",
    TransactionReturned: "transaction_returned",
    BankTransactionError: "bank_transaction_error",
    FiatAccountLimitExceeded: "fiat_account_limit_exceeded",
    InvalidBankAccountNumber: "invalid_bank_account_number",
    InvalidAchRtn: "invalid_ach_rtn",
    InvalidWireRtn: "invalid_wire_rtn",
    VendorInactive: "vendor_inactive"
};
