// @circle-fin/circle-sdk/dist/generated/models/reversal-reason.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReversalReason = void 0;
/**
 * Enumerated reason for a returned payment. Providing this reason in the request is recommended (to improve risk evaluation) but not required.
 * @export
 * @enum {string}
 */
exports.ReversalReason = {
    Duplicate: "duplicate",
    Fraudulent: "fraudulent",
    RequestedByCustomer: "requested_by_customer",
    BankTransactionError: "bank_transaction_error",
    InvalidAccountNumber: "invalid_account_number",
    InsufficientFunds: "insufficient_funds",
    PaymentStoppedByIssuer: "payment_stopped_by_issuer",
    PaymentReturned: "payment_returned",
    BankAccountIneligible: "bank_account_ineligible",
    InvalidAchRtn: "invalid_ach_rtn",
    UnauthorizedTransaction: "unauthorized_transaction",
    PaymentFailed: "payment_failed"
};
