// @circle-fin/circle-sdk/dist/generated/models/verification-error-code.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.VerificationErrorCode = void 0;
/**
 * Indicates the failure reason of the card verification. Only present on cards with failed verification. Possible values are [verification_failed, verification_fraud_detected, verification_denied, verification_not_supported_by_issuer, verification_stopped_by_issuer, card_failed, card_invalid, card_address_mismatch, card_zip_mismatch, card_cvv_invalid, card_expired, card_limit_violated, card_not_honored, card_cvv_required, credit_card_not_allowed, card_account_ineligible, card_network_unsupported]\'
 * @export
 * @enum {string}
 */
exports.VerificationErrorCode = {
    VerificationFailed: "verification_failed",
    VerificationFraudDetected: "verification_fraud_detected",
    VerificationDenied: "verification_denied",
    VerificationNotSupportedByIssuer: "verification_not_supported_by_issuer",
    VerificationStoppedByIssuer: "verification_stopped_by_issuer",
    CardFailed: "card_failed",
    CardInvalid: "card_invalid",
    CardAddressMismatch: "card_address_mismatch",
    CardZipMismatch: "card_zip_mismatch",
    CardCvvInvalid: "card_cvv_invalid",
    CardExpired: "card_expired",
    CardLimitViolated: "card_limit_violated",
    CardNotHonored: "card_not_honored",
    CardCvvRequired: "card_cvv_required",
    CreditCardNotAllowed: "credit_card_not_allowed",
    CardAccountIneligible: "card_account_ineligible",
    CardNetworkUnsupported: "card_network_unsupported"
};
