// @circle-fin/circle-sdk/dist/generated/apis/channels-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChannelsApi = exports.ChannelsApiFactory = exports.ChannelsApiFp = exports.ChannelsApiAxiosParamCreator = void 0;
const axios_1 = __importDefault(require("axios"));
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * ChannelsApi - axios parameter creator
 * @export
 */
const ChannelsApiAxiosParamCreator = function (configuration) {
    return {
        /**
         * Retrieve a list of channels with details (e.g. `cardDescriptor`, `achDescriptor`, etc.).
         * @summary List all channels
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        listChannels: (options = {}) => __awaiter(this, void 0, void 0, function* () {
            const localVarPath = `/v1/channels`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: "GET" }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearerAuth required
            // http bearer authentication required
            yield (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions
            };
        })
    };
};
exports.ChannelsApiAxiosParamCreator = ChannelsApiAxiosParamCreator;
/**
 * ChannelsApi - functional programming interface
 * @export
 */
const ChannelsApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.ChannelsApiAxiosParamCreator)(configuration);
    return {
        /**
         * Retrieve a list of channels with details (e.g. `cardDescriptor`, `achDescriptor`, etc.).
         * @summary List all channels
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        listChannels(options) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.listChannels(options);
                return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        }
    };
};
exports.ChannelsApiFp = ChannelsApiFp;
/**
 * ChannelsApi - factory interface
 * @export
 */
const ChannelsApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.ChannelsApiFp)(configuration);
    return {
        /**
         * Retrieve a list of channels with details (e.g. `cardDescriptor`, `achDescriptor`, etc.).
         * @summary List all channels
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        listChannels(options) {
            return localVarFp
                .listChannels(options)
                .then((request) => request(axios, basePath));
        }
    };
};
exports.ChannelsApiFactory = ChannelsApiFactory;
/**
 * ChannelsApi - object-oriented interface
 * @export
 * @class ChannelsApi
 * @extends {BaseAPI}
 */
class ChannelsApi extends base_1.BaseAPI {
    /**
     * Retrieve a list of channels with details (e.g. `cardDescriptor`, `achDescriptor`, etc.).
     * @summary List all channels
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof ChannelsApi
     */
    listChannels(options) {
        return (0, exports.ChannelsApiFp)(this.configuration)
            .listChannels(options)
            .then((request) => request(this.axios, this.basePath));
    }
}
exports.ChannelsApi = ChannelsApi;
