// @circle-fin/circle-sdk/dist/generated/apis/crypto-exchange-rates-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CryptoExchangeRatesApi = exports.CryptoExchangeRatesApiFactory = exports.CryptoExchangeRatesApiFp = exports.CryptoExchangeRatesApiAxiosParamCreator = void 0;
const axios_1 = __importDefault(require("axios"));
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * CryptoExchangeRatesApi - axios parameter creator
 * @export
 */
const CryptoExchangeRatesApiAxiosParamCreator = function (configuration) {
    return {
        /**
         * Fetch the current rates for the specified trading pair. The trading pair is defined by a base currency followed by a quote currency. The response contains buy and sell rates denominated in the quote currency. **Exchange rate is an estimate only and is subject to change by the time you submit the actual request.**
         * @summary Get a exchange rate
         * @param {'BTC-USD' | 'ETH-USD' | 'FLOW-USD'} tradingPair
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getExchangeRates: (tradingPair, options = {}) => __awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'tradingPair' is not null or undefined
            (0, common_1.assertParamExists)("getExchangeRates", "tradingPair", tradingPair);
            const localVarPath = `/v1/exchange/rates/{trading-pair}`.replace(`{${"trading-pair"}}`, encodeURIComponent(String(tradingPair)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: "GET" }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearerAuth required
            // http bearer authentication required
            yield (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions
            };
        })
    };
};
exports.CryptoExchangeRatesApiAxiosParamCreator = CryptoExchangeRatesApiAxiosParamCreator;
/**
 * CryptoExchangeRatesApi - functional programming interface
 * @export
 */
const CryptoExchangeRatesApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.CryptoExchangeRatesApiAxiosParamCreator)(configuration);
    return {
        /**
         * Fetch the current rates for the specified trading pair. The trading pair is defined by a base currency followed by a quote currency. The response contains buy and sell rates denominated in the quote currency. **Exchange rate is an estimate only and is subject to change by the time you submit the actual request.**
         * @summary Get a exchange rate
         * @param {'BTC-USD' | 'ETH-USD' | 'FLOW-USD'} tradingPair
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getExchangeRates(tradingPair, options) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.getExchangeRates(tradingPair, options);
                return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        }
    };
};
exports.CryptoExchangeRatesApiFp = CryptoExchangeRatesApiFp;
/**
 * CryptoExchangeRatesApi - factory interface
 * @export
 */
const CryptoExchangeRatesApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.CryptoExchangeRatesApiFp)(configuration);
    return {
        /**
         * Fetch the current rates for the specified trading pair. The trading pair is defined by a base currency followed by a quote currency. The response contains buy and sell rates denominated in the quote currency. **Exchange rate is an estimate only and is subject to change by the time you submit the actual request.**
         * @summary Get a exchange rate
         * @param {'BTC-USD' | 'ETH-USD' | 'FLOW-USD'} tradingPair
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getExchangeRates(tradingPair, options) {
            return localVarFp
                .getExchangeRates(tradingPair, options)
                .then((request) => request(axios, basePath));
        }
    };
};
exports.CryptoExchangeRatesApiFactory = CryptoExchangeRatesApiFactory;
/**
 * CryptoExchangeRatesApi - object-oriented interface
 * @export
 * @class CryptoExchangeRatesApi
 * @extends {BaseAPI}
 */
class CryptoExchangeRatesApi extends base_1.BaseAPI {
    /**
     * Fetch the current rates for the specified trading pair. The trading pair is defined by a base currency followed by a quote currency. The response contains buy and sell rates denominated in the quote currency. **Exchange rate is an estimate only and is subject to change by the time you submit the actual request.**
     * @summary Get a exchange rate
     * @param {'BTC-USD' | 'ETH-USD' | 'FLOW-USD'} tradingPair
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof CryptoExchangeRatesApi
     */
    getExchangeRates(tradingPair, options) {
        return (0, exports.CryptoExchangeRatesApiFp)(this.configuration)
            .getExchangeRates(tradingPair, options)
            .then((request) => request(this.axios, this.basePath));
    }
}
exports.CryptoExchangeRatesApi = CryptoExchangeRatesApi;
