// @circle-fin/circle-sdk/dist/generated/apis/crypto-address-book-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CryptoAddressBookApi = exports.CryptoAddressBookApiFactory = exports.CryptoAddressBookApiFp = exports.CryptoAddressBookApiAxiosParamCreator = void 0;
const axios_1 = __importDefault(require("axios"));
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * CryptoAddressBookApi - axios parameter creator
 * @export
 */
const CryptoAddressBookApiAxiosParamCreator = function (configuration) {
    return {
        /**
         *
         * @summary Create a recipient
         * @param {AddressBookRecipientRequest} [addressBookRecipientRequest]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        createAddressBookRecipient: (addressBookRecipientRequest, options = {}) => __awaiter(this, void 0, void 0, function* () {
            const localVarPath = `/v1/addressBook/recipients`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: "POST" }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearerAuth required
            // http bearer authentication required
            yield (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            localVarHeaderParameter["Content-Type"] = "application/json";
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(addressBookRecipientRequest, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions
            };
        }),
        /**
         *
         * @summary Delete a recipient
         * @param {string} id Universally unique identifier (UUID v4) of a resource.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        deleteAddressBookRecipient: (id, options = {}) => __awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)("deleteAddressBookRecipient", "id", id);
            const localVarPath = `/v1/addressBook/recipients/{id}`.replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: "DELETE" }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearerAuth required
            // http bearer authentication required
            yield (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions
            };
        }),
        /**
         *
         * @summary Get a recipient
         * @param {string} id Universally unique identifier (UUID v4) of a resource.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getAddressBookRecipient: (id, options = {}) => __awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)("getAddressBookRecipient", "id", id);
            const localVarPath = `/v1/addressBook/recipients/{id}`.replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: "GET" }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearerAuth required
            // http bearer authentication required
            yield (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions
            };
        }),
        /**
         *
         * @summary List all recipients
         * @param {string} [address] Filters results to fetch only address book recipients with the provided address.
         * @param {string} [chain] Filters results to fetch only address book recipients with the provided chain.
         * @param {string} [email] Filters results to fetch only address book recipients that have the provided email in their metadata.
         * @param {'pending' | 'inactive' | 'active' | 'denied'} [status] Filters results to fetch only address book recipients that have the provided status.
         * @param {string} [from] Queries items created since the specified date-time (inclusive).
         * @param {string} [to] Queries items created before the specified date-time (inclusive).
         * @param {string} [pageBefore] A collection ID value used for pagination.  It marks the exclusive end of a page. When provided, the collection resource will return the next &#x60;n&#x60; items before the id, with &#x60;n&#x60; being specified by &#x60;pageSize&#x60;.  The items will be returned in the natural order of the collection.  The resource will return the first page if neither &#x60;pageAfter&#x60; nor &#x60;pageBefore&#x60; are specified.  SHOULD NOT be used in conjuction with pageAfter.
         * @param {string} [pageAfter] A collection ID value used for pagination.  It marks the exclusive begin of a page. When provided, the collection resource will return the next &#x60;n&#x60; items after the id, with &#x60;n&#x60; being specified by &#x60;pageSize&#x60;.  The items will be returned in the natural order of the collection.  The resource will return the first page if neither &#x60;pageAfter&#x60; nor &#x60;pageBefore&#x60; are specified.  SHOULD NOT be used in conjuction with pageBefore.
         * @param {number} [pageSize] Limits the number of items to be returned.  Some collections have a strict upper bound that will disregard this value. In case the specified value is higher than the allowed limit, the collection limit will be used.  If avoided, the collection will determine the page size itself.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        listAddressBookRecipients: (address, chain, email, status, from, to, pageBefore, pageAfter, pageSize, options = {}) => __awaiter(this, void 0, void 0, function* () {
            const localVarPath = `/v1/addressBook/recipients`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: "GET" }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearerAuth required
            // http bearer authentication required
            yield (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            if (address !== undefined) {
                localVarQueryParameter["address"] = address;
            }
            if (chain !== undefined) {
                localVarQueryParameter["chain"] = chain;
            }
            if (email !== undefined) {
                localVarQueryParameter["email"] = email;
            }
            if (status !== undefined) {
                localVarQueryParameter["status"] = status;
            }
            if (from !== undefined) {
                localVarQueryParameter["from"] =
                    from instanceof Date ? from.toISOString() : from;
            }
            if (to !== undefined) {
                localVarQueryParameter["to"] =
                    to instanceof Date ? to.toISOString() : to;
            }
            if (pageBefore !== undefined) {
                localVarQueryParameter["pageBefore"] = pageBefore;
            }
            if (pageAfter !== undefined) {
                localVarQueryParameter["pageAfter"] = pageAfter;
            }
            if (pageSize !== undefined) {
                localVarQueryParameter["pageSize"] = pageSize;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions
            };
        }),
        /**
         *
         * @summary Modify a recipient
         * @param {string} id Universally unique identifier (UUID v4) of a resource.
         * @param {AddressBookRecipientModifyRequest} [addressBookRecipientModifyRequest]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        modifyAddressBookRecipient: (id, addressBookRecipientModifyRequest, options = {}) => __awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)("modifyAddressBookRecipient", "id", id);
            const localVarPath = `/v1/addressBook/recipients/{id}`.replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: "PATCH" }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearerAuth required
            // http bearer authentication required
            yield (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            localVarHeaderParameter["Content-Type"] = "application/json";
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(addressBookRecipientModifyRequest, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions
            };
        })
    };
};
exports.CryptoAddressBookApiAxiosParamCreator = CryptoAddressBookApiAxiosParamCreator;
/**
 * CryptoAddressBookApi - functional programming interface
 * @export
 */
const CryptoAddressBookApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.CryptoAddressBookApiAxiosParamCreator)(configuration);
    return {
        /**
         *
         * @summary Create a recipient
         * @param {AddressBookRecipientRequest} [addressBookRecipientRequest]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        createAddressBookRecipient(addressBookRecipientRequest, options) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.createAddressBookRecipient(addressBookRecipientRequest, options);
                return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
        /**
         *
         * @summary Delete a recipient
         * @param {string} id Universally unique identifier (UUID v4) of a resource.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        deleteAddressBookRecipient(id, options) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.deleteAddressBookRecipient(id, options);
                return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
        /**
         *
         * @summary Get a recipient
         * @param {string} id Universally unique identifier (UUID v4) of a resource.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getAddressBookRecipient(id, options) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.getAddressBookRecipient(id, options);
                return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
        /**
         *
         * @summary List all recipients
         * @param {string} [address] Filters results to fetch only address book recipients with the provided address.
         * @param {string} [chain] Filters results to fetch only address book recipients with the provided chain.
         * @param {string} [email] Filters results to fetch only address book recipients that have the provided email in their metadata.
         * @param {'pending' | 'inactive' | 'active' | 'denied'} [status] Filters results to fetch only address book recipients that have the provided status.
         * @param {string} [from] Queries items created since the specified date-time (inclusive).
         * @param {string} [to] Queries items created before the specified date-time (inclusive).
         * @param {string} [pageBefore] A collection ID value used for pagination.  It marks the exclusive end of a page. When provided, the collection resource will return the next &#x60;n&#x60; items before the id, with &#x60;n&#x60; being specified by &#x60;pageSize&#x60;.  The items will be returned in the natural order of the collection.  The resource will return the first page if neither &#x60;pageAfter&#x60; nor &#x60;pageBefore&#x60; are specified.  SHOULD NOT be used in conjuction with pageAfter.
         * @param {string} [pageAfter] A collection ID value used for pagination.  It marks the exclusive begin of a page. When provided, the collection resource will return the next &#x60;n&#x60; items after the id, with &#x60;n&#x60; being specified by &#x60;pageSize&#x60;.  The items will be returned in the natural order of the collection.  The resource will return the first page if neither &#x60;pageAfter&#x60; nor &#x60;pageBefore&#x60; are specified.  SHOULD NOT be used in conjuction with pageBefore.
         * @param {number} [pageSize] Limits the number of items to be returned.  Some collections have a strict upper bound that will disregard this value. In case the specified value is higher than the allowed limit, the collection limit will be used.  If avoided, the collection will determine the page size itself.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        listAddressBookRecipients(address, chain, email, status, from, to, pageBefore, pageAfter, pageSize, options) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.listAddressBookRecipients(address, chain, email, status, from, to, pageBefore, pageAfter, pageSize, options);
                return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
        /**
         *
         * @summary Modify a recipient
         * @param {string} id Universally unique identifier (UUID v4) of a resource.
         * @param {AddressBookRecipientModifyRequest} [addressBookRecipientModifyRequest]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        modifyAddressBookRecipient(id, addressBookRecipientModifyRequest, options) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.modifyAddressBookRecipient(id, addressBookRecipientModifyRequest, options);
                return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        }
    };
};
exports.CryptoAddressBookApiFp = CryptoAddressBookApiFp;
/**
 * CryptoAddressBookApi - factory interface
 * @export
 */
const CryptoAddressBookApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.CryptoAddressBookApiFp)(configuration);
    return {
        /**
         *
         * @summary Create a recipient
         * @param {AddressBookRecipientRequest} [addressBookRecipientRequest]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        createAddressBookRecipient(addressBookRecipientRequest, options) {
            return localVarFp
                .createAddressBookRecipient(addressBookRecipientRequest, options)
                .then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Delete a recipient
         * @param {string} id Universally unique identifier (UUID v4) of a resource.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        deleteAddressBookRecipient(id, options) {
            return localVarFp
                .deleteAddressBookRecipient(id, options)
                .then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Get a recipient
         * @param {string} id Universally unique identifier (UUID v4) of a resource.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getAddressBookRecipient(id, options) {
            return localVarFp
                .getAddressBookRecipient(id, options)
                .then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary List all recipients
         * @param {string} [address] Filters results to fetch only address book recipients with the provided address.
         * @param {string} [chain] Filters results to fetch only address book recipients with the provided chain.
         * @param {string} [email] Filters results to fetch only address book recipients that have the provided email in their metadata.
         * @param {'pending' | 'inactive' | 'active' | 'denied'} [status] Filters results to fetch only address book recipients that have the provided status.
         * @param {string} [from] Queries items created since the specified date-time (inclusive).
         * @param {string} [to] Queries items created before the specified date-time (inclusive).
         * @param {string} [pageBefore] A collection ID value used for pagination.  It marks the exclusive end of a page. When provided, the collection resource will return the next &#x60;n&#x60; items before the id, with &#x60;n&#x60; being specified by &#x60;pageSize&#x60;.  The items will be returned in the natural order of the collection.  The resource will return the first page if neither &#x60;pageAfter&#x60; nor &#x60;pageBefore&#x60; are specified.  SHOULD NOT be used in conjuction with pageAfter.
         * @param {string} [pageAfter] A collection ID value used for pagination.  It marks the exclusive begin of a page. When provided, the collection resource will return the next &#x60;n&#x60; items after the id, with &#x60;n&#x60; being specified by &#x60;pageSize&#x60;.  The items will be returned in the natural order of the collection.  The resource will return the first page if neither &#x60;pageAfter&#x60; nor &#x60;pageBefore&#x60; are specified.  SHOULD NOT be used in conjuction with pageBefore.
         * @param {number} [pageSize] Limits the number of items to be returned.  Some collections have a strict upper bound that will disregard this value. In case the specified value is higher than the allowed limit, the collection limit will be used.  If avoided, the collection will determine the page size itself.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        listAddressBookRecipients(address, chain, email, status, from, to, pageBefore, pageAfter, pageSize, options) {
            return localVarFp
                .listAddressBookRecipients(address, chain, email, status, from, to, pageBefore, pageAfter, pageSize, options)
                .then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Modify a recipient
         * @param {string} id Universally unique identifier (UUID v4) of a resource.
         * @param {AddressBookRecipientModifyRequest} [addressBookRecipientModifyRequest]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        modifyAddressBookRecipient(id, addressBookRecipientModifyRequest, options) {
            return localVarFp
                .modifyAddressBookRecipient(id, addressBookRecipientModifyRequest, options)
                .then((request) => request(axios, basePath));
        }
    };
};
exports.CryptoAddressBookApiFactory = CryptoAddressBookApiFactory;
/**
 * CryptoAddressBookApi - object-oriented interface
 * @export
 * @class CryptoAddressBookApi
 * @extends {BaseAPI}
 */
class CryptoAddressBookApi extends base_1.BaseAPI {
    /**
     *
     * @summary Create a recipient
     * @param {AddressBookRecipientRequest} [addressBookRecipientRequest]
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof CryptoAddressBookApi
     */
    createAddressBookRecipient(addressBookRecipientRequest, options) {
        return (0, exports.CryptoAddressBookApiFp)(this.configuration)
            .createAddressBookRecipient(addressBookRecipientRequest, options)
            .then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Delete a recipient
     * @param {string} id Universally unique identifier (UUID v4) of a resource.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof CryptoAddressBookApi
     */
    deleteAddressBookRecipient(id, options) {
        return (0, exports.CryptoAddressBookApiFp)(this.configuration)
            .deleteAddressBookRecipient(id, options)
            .then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Get a recipient
     * @param {string} id Universally unique identifier (UUID v4) of a resource.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof CryptoAddressBookApi
     */
    getAddressBookRecipient(id, options) {
        return (0, exports.CryptoAddressBookApiFp)(this.configuration)
            .getAddressBookRecipient(id, options)
            .then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary List all recipients
     * @param {string} [address] Filters results to fetch only address book recipients with the provided address.
     * @param {string} [chain] Filters results to fetch only address book recipients with the provided chain.
     * @param {string} [email] Filters results to fetch only address book recipients that have the provided email in their metadata.
     * @param {'pending' | 'inactive' | 'active' | 'denied'} [status] Filters results to fetch only address book recipients that have the provided status.
     * @param {string} [from] Queries items created since the specified date-time (inclusive).
     * @param {string} [to] Queries items created before the specified date-time (inclusive).
     * @param {string} [pageBefore] A collection ID value used for pagination.  It marks the exclusive end of a page. When provided, the collection resource will return the next &#x60;n&#x60; items before the id, with &#x60;n&#x60; being specified by &#x60;pageSize&#x60;.  The items will be returned in the natural order of the collection.  The resource will return the first page if neither &#x60;pageAfter&#x60; nor &#x60;pageBefore&#x60; are specified.  SHOULD NOT be used in conjuction with pageAfter.
     * @param {string} [pageAfter] A collection ID value used for pagination.  It marks the exclusive begin of a page. When provided, the collection resource will return the next &#x60;n&#x60; items after the id, with &#x60;n&#x60; being specified by &#x60;pageSize&#x60;.  The items will be returned in the natural order of the collection.  The resource will return the first page if neither &#x60;pageAfter&#x60; nor &#x60;pageBefore&#x60; are specified.  SHOULD NOT be used in conjuction with pageBefore.
     * @param {number} [pageSize] Limits the number of items to be returned.  Some collections have a strict upper bound that will disregard this value. In case the specified value is higher than the allowed limit, the collection limit will be used.  If avoided, the collection will determine the page size itself.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof CryptoAddressBookApi
     */
    listAddressBookRecipients(address, chain, email, status, from, to, pageBefore, pageAfter, pageSize, options) {
        return (0, exports.CryptoAddressBookApiFp)(this.configuration)
            .listAddressBookRecipients(address, chain, email, status, from, to, pageBefore, pageAfter, pageSize, options)
            .then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Modify a recipient
     * @param {string} id Universally unique identifier (UUID v4) of a resource.
     * @param {AddressBookRecipientModifyRequest} [addressBookRecipientModifyRequest]
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof CryptoAddressBookApi
     */
    modifyAddressBookRecipient(id, addressBookRecipientModifyRequest, options) {
        return (0, exports.CryptoAddressBookApiFp)(this.configuration)
            .modifyAddressBookRecipient(id, addressBookRecipientModifyRequest, options)
            .then((request) => request(this.axios, this.basePath));
    }
}
exports.CryptoAddressBookApi = CryptoAddressBookApi;
