// @circle-fin/circle-sdk/dist/generated/apis/pull-crypto-payments-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PullCryptoPaymentsApi = exports.PullCryptoPaymentsApiFactory = exports.PullCryptoPaymentsApiFp = exports.PullCryptoPaymentsApiAxiosParamCreator = void 0;
const axios_1 = __importDefault(require("axios"));
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * PullCryptoPaymentsApi - axios parameter creator
 * @export
 */
const PullCryptoPaymentsApiAxiosParamCreator = function (configuration) {
    return {
        /**
         *
         * @summary Create a crypto payment
         * @param {CryptoPaymentCreationRequest} [cryptoPaymentCreationRequest]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        createCryptoPayment: (cryptoPaymentCreationRequest, options = {}) => __awaiter(this, void 0, void 0, function* () {
            const localVarPath = `/v1/payments/crypto`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: "POST" }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearerAuth required
            // http bearer authentication required
            yield (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            localVarHeaderParameter["Content-Type"] = "application/json";
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(cryptoPaymentCreationRequest, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions
            };
        }),
        /**
         *
         * @summary Get a typed message for signing
         * @param {string} endUserAddress User wallet address
         * @param {string} paymentIntentId Payment intent id which is from the create payment intent endpoint response
         * @param {string} [amount]
         * @param {'USD'} [currency] Only support USD at the moment
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        presign: (endUserAddress, paymentIntentId, amount, currency, options = {}) => __awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'endUserAddress' is not null or undefined
            (0, common_1.assertParamExists)("presign", "endUserAddress", endUserAddress);
            // verify required parameter 'paymentIntentId' is not null or undefined
            (0, common_1.assertParamExists)("presign", "paymentIntentId", paymentIntentId);
            const localVarPath = `/v1/payments/presign`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: "GET" }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearerAuth required
            // http bearer authentication required
            yield (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            if (endUserAddress !== undefined) {
                localVarQueryParameter["endUserAddress"] = endUserAddress;
            }
            if (paymentIntentId !== undefined) {
                localVarQueryParameter["paymentIntentId"] = paymentIntentId;
            }
            if (amount !== undefined) {
                localVarQueryParameter["amount"] = amount;
            }
            if (currency !== undefined) {
                localVarQueryParameter["currency"] = currency;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions
            };
        })
    };
};
exports.PullCryptoPaymentsApiAxiosParamCreator = PullCryptoPaymentsApiAxiosParamCreator;
/**
 * PullCryptoPaymentsApi - functional programming interface
 * @export
 */
const PullCryptoPaymentsApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.PullCryptoPaymentsApiAxiosParamCreator)(configuration);
    return {
        /**
         *
         * @summary Create a crypto payment
         * @param {CryptoPaymentCreationRequest} [cryptoPaymentCreationRequest]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        createCryptoPayment(cryptoPaymentCreationRequest, options) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.createCryptoPayment(cryptoPaymentCreationRequest, options);
                return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
        /**
         *
         * @summary Get a typed message for signing
         * @param {string} endUserAddress User wallet address
         * @param {string} paymentIntentId Payment intent id which is from the create payment intent endpoint response
         * @param {string} [amount]
         * @param {'USD'} [currency] Only support USD at the moment
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        presign(endUserAddress, paymentIntentId, amount, currency, options) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.presign(endUserAddress, paymentIntentId, amount, currency, options);
                return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        }
    };
};
exports.PullCryptoPaymentsApiFp = PullCryptoPaymentsApiFp;
/**
 * PullCryptoPaymentsApi - factory interface
 * @export
 */
const PullCryptoPaymentsApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.PullCryptoPaymentsApiFp)(configuration);
    return {
        /**
         *
         * @summary Create a crypto payment
         * @param {CryptoPaymentCreationRequest} [cryptoPaymentCreationRequest]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        createCryptoPayment(cryptoPaymentCreationRequest, options) {
            return localVarFp
                .createCryptoPayment(cryptoPaymentCreationRequest, options)
                .then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Get a typed message for signing
         * @param {string} endUserAddress User wallet address
         * @param {string} paymentIntentId Payment intent id which is from the create payment intent endpoint response
         * @param {string} [amount]
         * @param {'USD'} [currency] Only support USD at the moment
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        presign(endUserAddress, paymentIntentId, amount, currency, options) {
            return localVarFp
                .presign(endUserAddress, paymentIntentId, amount, currency, options)
                .then((request) => request(axios, basePath));
        }
    };
};
exports.PullCryptoPaymentsApiFactory = PullCryptoPaymentsApiFactory;
/**
 * PullCryptoPaymentsApi - object-oriented interface
 * @export
 * @class PullCryptoPaymentsApi
 * @extends {BaseAPI}
 */
class PullCryptoPaymentsApi extends base_1.BaseAPI {
    /**
     *
     * @summary Create a crypto payment
     * @param {CryptoPaymentCreationRequest} [cryptoPaymentCreationRequest]
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof PullCryptoPaymentsApi
     */
    createCryptoPayment(cryptoPaymentCreationRequest, options) {
        return (0, exports.PullCryptoPaymentsApiFp)(this.configuration)
            .createCryptoPayment(cryptoPaymentCreationRequest, options)
            .then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Get a typed message for signing
     * @param {string} endUserAddress User wallet address
     * @param {string} paymentIntentId Payment intent id which is from the create payment intent endpoint response
     * @param {string} [amount]
     * @param {'USD'} [currency] Only support USD at the moment
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof PullCryptoPaymentsApi
     */
    presign(endUserAddress, paymentIntentId, amount, currency, options) {
        return (0, exports.PullCryptoPaymentsApiFp)(this.configuration)
            .presign(endUserAddress, paymentIntentId, amount, currency, options)
            .then((request) => request(this.axios, this.basePath));
    }
}
exports.PullCryptoPaymentsApi = PullCryptoPaymentsApi;
